### DISCUSSION TITLE
How to write a parser where AST nodes hold reference to source string?

### DISCUSSION THREAD

**Discussion by brundonsmith** [zesterer/chumsky#147] [Q&A] [answered]
I'd like various AST nodes to hold references directly to the original source string (instead of allocating their own copies of all those substrings of it). Is this possible with chumsky?

**Comment by zesterer** [comment, marked as answer]
It's currently not possible, but will become possible once the `zero-copy` branch has been merged.

**Reply by brundonsmith** [reply]
Awesome! Thanks for the quick response
